tfhe/src/integer/server_key/radix_parallel/mul.rs

```diff
@@ -273,6 +273,148 @@ impl ServerKey {
         *ct1 = self.smart_block_mul_parallelized(ct1, ct2, index);
     }
 
+    /// Sums the terms, putting the result into lhs
+    ///
+    /// This sums all of the terms in `terms` and overwrites
+    /// `lhs` with the result.
+    ///
+    /// Each of the input term is expected to have _at most_ one carry consumed
+    pub(crate) fn sum_multiplication_terms_into<PBSOrder: PBSOrderMarker>(
+        &self,
+        lhs: &mut RadixCiphertext<PBSOrder>,
+        mut terms: Vec<RadixCiphertext<PBSOrder>>,
+    ) {
+        let num_blocks = lhs.blocks.len();
+
+        let num_bits_in_carry = self.key.carry_modulus.0.ilog2() as u64;
+        // Among those bits of carry, we know one is already consumed by
+        // the last step of the unchecked_block_mul_parallelized
+        // for each of the term.
+        //
+        // It means we can only do num_bits_in_carry - 1 additions
+        // Which then means we can iter on chunks of num_bits_in_carry
+        // to add themp up as it will result in num_bits_in_carry - 1 additions
+        let chunk_size = num_bits_in_carry as usize;
+
+        // For the last chunk, we want to finish it off
+        // using an addition that does not leave the resulting ciphertext with
+        // non empty carries.
+        //
```

Comment:
`we want to use an addition that leaves the resulting ciphertext with empty carries`

seems more readable if I got what you wanted to say

Reply:
yep my phrasing was a bit confusing

Reply:
looks like this was not updated, let's keep it for a next PR